src/faction_camp.cpp

```diff
@@ -2385,43 +2375,40 @@ void talk_function::camp_hunting_results( int skill, const std::string &task, in
 }
 
 // map manipulation functions
-bool talk_function::om_camp_upgrade( npc &comp, const tripoint &omt_pos )
+bool basecamp::om_upgrade( npc &comp, const std::string &next_upgrade, const tripoint &upos )
 {
     editmap edit;
 
-    oter_id &omt_ref = overmap_buffer.ter( omt_pos );
-    std::string om_old = omt_ref.id().c_str();
-
-    if( !edit.mapgen_set( om_next_upgrade( om_old ), tripoint( omt_pos ) ) ) {
+    if( !edit.mapgen_set( next_upgrade, upos ) ) {
         return false;
     }
 
-    tripoint np = tripoint( 5 * SEEX + 5 % SEEX, 4 * SEEY + 5 % SEEY, comp.posz() );
-    np.x = g->u.posx() + 1;
-    np.y = g->u.posy();
+    if( upos == pos ) {
+        tripoint np = tripoint( 5 * SEEX + 5 % SEEX, 4 * SEEY + 5 % SEEY, pos.z );
+        np.x = g->u.posx() + 1;
+        np.y = g->u.posy();
 
-    tripoint p( 0, 0, comp.posz() );
-    int &x = p.x;
-    int &y = p.y;
-    for( x = 0; x < SEEX * g->m.getmapsize(); x++ ) {
-        for( y = 0; y < SEEY * g->m.getmapsize(); y++ ) {
-            if( g->m.ter( x, y ) == ter_id( "t_floor_green" ) ) {
-                np.x = x;
-                np.y = y;
+        tripoint p( 0, 0, comp.posz() );
+        int &x = p.x;
+        int &y = p.y;
+        for( x = 0; x < SEEX * g->m.getmapsize(); x++ ) {
+            for( y = 0; y < SEEY * g->m.getmapsize(); y++ ) {
+                if( g->m.ter( x, y ) == ter_id( "t_floor_green" ) ) {
+                    np.x = x;
+                    np.y = y;
+                }
             }
         }
+        comp.setpos( np );
+        comp.set_destination();
     }
-    comp.setpos( np );
-    comp.set_destination();
     g->load_npcs();
     return true;
 }
 
 int talk_function::om_harvest_furn( npc &comp, const tripoint &omt_tgt, const furn_id &f,
                                     float chance, bool force_bash )
 {
-    oter_id &omt_ref = overmap_buffer.ter( omt_tgt );
-    omt_ref = oter_id( omt_ref.id().c_str() );
```

Comment:
Very nice find. I wonder what the original intention was for that.

Reply:
Probably a copy-paste error.